src/common/utils/map_iter.h

```diff
@@ -23,13 +23,12 @@
 
 namespace raksha::utils {
 
-template <typename T, typename U, typename F>
-std::vector<U> MapIter(const std::vector<T>& input, F f) {
+template <typename U, typename C, typename F>
+std::vector<U> MapIter(const C& input, F map_fn) {
```

Comment:
It looks like now that the first argument is a reference to C, it works for other kinds of collections of type C? It's a bit non-obvious to me that this works. Is it worth putting a comment that the intention is for the first argument to be some kind of collection?

Reply:
Good point. Done.